File: Glucloser/app/src/main/java/com/nlefler/glucloser/models/BolusEvent.java
package com.nlefler.glucloser.models;

import java.util.Date;
import java.util.List;

import io.realm.RealmList;

/**
 * Created by Nathan Lefler on 5/8/15.
 */
public interface BolusEvent {
    public String getId();
    public void setId(String id);

    public Date getDate();
    public void setDate(Date date);

    public int getCarbs();
    public void setCarbs(int carbs);

    public float getInsulin();
    public void setInsulin(float insulin);

    public BloodSugar getBeforeSugar();
    public void setBeforeSugar(BloodSugar beforeSugar);

    public boolean isCorrection();
    public void setCorrection(boolean isCorrection);

    public RealmList<Food> getFoods();
    public void setFoods(RealmList<Food> foods);
}

File: Glucloser/app/src/main/java/com/nlefler/glucloser/models/BolusEvent.kt
package com.nlefler.glucloser.models

import java.util.Date

import io.realm.RealmList

/**
 * Created by Nathan Lefler on 5/8/15.
 */
public interface BolusEvent {
    var id: String
    var date: Date
    var bolusPattern: BolusPattern
    var carbs: Int
    var insulin: Float
    var beforeSugar: BloodSugar?
    var isCorrection: Boolean
    var foods: RealmList<Food>
}
